File: tools/sdk-go/staffkm/quota.go
package staffkm

// QuotaSummary fetches the current workspace's quota usage.
func (c *Client) QuotaSummary() (*QuotaSummary, error) {
	req, err := c.newRequest("GET", "/api/v1/quota/summary", nil)
	if err != nil {
		return nil, err
	}
	var resp struct {
		Data QuotaSummary `json:"data"`
	}
	if err := c.do(req, &resp); err != nil {
		return nil, err
	}
	return &resp.Data, nil
}

// SetQuota updates monthly caps. Pass nil pointers to leave a field unchanged.
func (c *Client) SetQuota(monthlyTokenCap *int, monthlyCostCapUSD *float64) (*QuotaSummary, error) {
	body := map[string]interface{}{
		"monthly_token_cap":    monthlyTokenCap,
		"monthly_cost_cap_usd": monthlyCostCapUSD,
	}
	req, err := c.newRequest("PUT", "/api/v1/quota", body)
	if err != nil {
		return nil, err
	}
	var resp struct {
		Data QuotaSummary `json:"data"`
	}
	if err := c.do(req, &resp); err != nil {
		return nil, err
	}
	return &resp.Data, nil
}

// ListQuotas (admin) returns all workspace quotas.
func (c *Client) ListQuotas() ([]map[string]interface{}, error) {
	req, err := c.newRequest("GET", "/api/v1/admin/quota", nil)
	if err != nil {
		return nil, err
	}
	var resp struct {
		Data []map[string]interface{} `json:"data"`
	}
	if err := c.do(req, &resp); err != nil {
		return nil, err
	}
	return resp.Data, nil
}

File: packages/python/staffkm-sdk/staffkm/resources/quota.py
"""Quota resource."""
from __future__ import annotations

import httpx


class QuotaResource:
    def __init__(self, http: httpx.Client):
        self._http = http

    def summary(self) -> dict:
        r = self._http.get("/api/v1/quota/summary")
        r.raise_for_status()
        return r.json().get("data") or r.json()

    def set(
        self,
        *,
        monthly_token_cap: int | None = None,
        monthly_cost_cap_usd: float | None = None,
    ) -> dict:
        body = {
            "monthly_token_cap": monthly_token_cap,
            "monthly_cost_cap_usd": monthly_cost_cap_usd,
        }
        r = self._http.put("/api/v1/quota", json=body)
        r.raise_for_status()
        return r.json().get("data") or r.json()

    def list(self) -> list[dict]:
        """Admin: list all workspace quotas."""
        r = self._http.get("/api/v1/admin/quota")
        r.raise_for_status()
        return r.json().get("data") or []
